brave/browser-laptop
# SwitchControl text now toggles checked state, and is consistently used

### Test Plan
- Open the _Clear Browsing Data_ model or the _Brave_ extension panel
  - verify that clicking a switch label causes the checked state to toggle
  - verify no visual regressions with switches and text
- Open preferences
  - test switch controls and text labels on all tabs and dialogs
- Open a new tab and navigate to any Url
- Open the 'find in page' dialog
  - check the 'case sensitivity' switch
  - verify that the switch and label are operating the toggle
  - verify that the tab theme colors are being used for the switch text label

# Supporting text click on switchControl

The click event will now fire from the label, whether it is to the left, right, or top of the switch element.

This also converts text from `<span />` to `<label />` for accessibility and more appropriate renderer defaults (the cursor) to convey that clicking the label will toggle the checked state.

Usages this affects:
 - all items in the _Brave_ extension panel (compact and full mode)
 - the _Clear Browsing Data_ model
 - the _MessageBox_ component
 - about:adblock

This is a partial resolution only for #8243 since many usages do not use the built-in switchControl label element but render their own text separately. Changing some of those usages to the switchControl label element causes some minor visual regression which may warrant some discussion. A subsequent PR will address converting these usages, which are not covered by this PR:
 - most switches in all sections of about:preferences
 - case-sensitivity switch in the findPage dialog

# Ensuring SwitchControl text prop is provided across usages



This ensures all remaining uses of `SwitchControl` gain the built-in text label click handler.

Covers:
- the case-sensitivity toggle in findbar
- the switches on all the sections of about:preferences page

I refactored a lot of the `SettingCheckbox` component as it was duplicating quite a bit of the `SwitchControl` styles even though it was including the common component, which was creating style conflicts when using the text label feature of `SwitchControl` - a result of integrating the fix for #8243.

I created a separate PR for this as it involves changes to multiple components which consumed `SwitchControl` in different ways. If that is too granular then I'm happy to combine these changes with #10468. 

I'm aware of #7321 which calls for a larger refactor of switchControl and can be a good opportunity to create a cleaner version of this component whilst analyzing the different state variations now required from the varied uses I've come across. I made sure usages of the component are more consistent, but did not attempt a clean-up of the existing version of the component because of that upcoming issue.

Submitter Checklist:


Reviewer Checklist:

Tests
